### PR TITLE
chore: test coverage improvement

### PR DESCRIPTION
**Pull request by VictorAssunc**
I noticed that some functions and function aliases weren't being completely covered by the tests, so I decided to create and improve some tests.

I decided to separate by commits instead of PRs to not spam the repository.

I'm not sure if I chose the best approach in each test, so I'm willing to change and learn more :)

### CODE REVIEW

File: parallel/slice_test.go [hunk 1]
```diff
@@ -24,9 +24,21 @@ func TestMap(t *testing.T) {
 	is.Equal(result2, []string{"1", "2", "3", "4"})
 }
 
-func TestTimes(t *testing.T) {
+func TestForEach(t *testing.T) {
 	is := assert.New(t)
+	
+	count := 0
```
Comment (samber) [inline parallel/slice_test.go:30]: please use 

```go
var counter uint64
atomic.AddUint64(&counter, 1)
```

instead

**Review by samber — CHANGES_REQUESTED** [review]
hi @VictorAssunc and thanks for this first contrib

i made 1 comment

File: parallel/slice_test.go (hunk 1 above)
Comment (VictorAssunc) [reply parallel/slice_test.go:30]: ok, i will change it

Comment (VictorAssunc) [reply parallel/slice_test.go]: done ♻️

**Review by samber — APPROVED** [review]

**Comment by samber** [comment]
I removed the parallel execution when you use a custom rand seed ;)

**Pull request merged**